hbase-client/src/main/java/org/apache/hadoop/hbase/security/oauthbearer/internals/OAuthBearerSaslClientProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.security.oauthbearer.internals;
+
+import static org.apache.hadoop.hbase.security.token.OAuthBearerTokenUtil.OAUTHBEARER_MECHANISM;
+import java.security.Provider;
+import java.security.Security;
+import org.apache.yetus.audience.InterfaceAudience;
+
+@InterfaceAudience.Public
+public class OAuthBearerSaslClientProvider extends Provider {
+  private static final long serialVersionUID = 1L;
+
+  protected OAuthBearerSaslClientProvider() {
+    super("SASL/OAUTHBEARER Client Provider", 1.0, "SASL/OAUTHBEARER Client Provider for HBase");
+    put("SaslClientFactory." + OAUTHBEARER_MECHANISM,
+      OAuthBearerSaslClient.OAuthBearerSaslClientFactory.class.getName());
+  }
+
+  public static void initialize() {
+    Security.addProvider(new OAuthBearerSaslClientProvider());
```

Comment:
It doesn't work without it, because we don't instantiate the client and the server directly, but asking the Sasl framework to create them for us. In the client auth provider we call `Sasl.createSaslClient()` to create the client which doesn't work without the client registered for the mechanism.

Reply:
> In the client auth provider we call Sasl.createSaslClient() to create the client which doesn't work without the client registered for the mechanism.

Ah, ok. I remember that when we were messing around with other custom Providers. Thanks for clarifying!